frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature

```diff
@@ -130,3 +130,81 @@ Feature: Create the pipeline from builder page
               And user will see workspace mentioned as "git-opt (optional)" in the Workspaces section of Pipeline Details page
+
+
+        @regression, @manual
+        Scenario: Add finally task node
+            Given user is at Pipeline Builder page
+             When user navigates to task section
+              And user clicks on Add finally task
+              And user clicks on Select task
+              And user select "git-cli"
+              And user clicks on Add finally task again
+              And user clicks on Select task
+              And user selects "git-clone"
+             Then user sees Add finally task bubble is attached to the Select Task
+              And user sees "git-cli" and "git-clone" tasks in parallel
+              And user sees "Add finally task" option below "git-clone" task
+
+
+        @regression
+        Scenario: Create a pipeline with finally task node
+            Given user is at Pipeline Builder page
+             When user enters pipeline name
+              And user selects the first task as "git-clone"
+              And user clicks on Add finally task
+              And user clicks on Select task
+              And user select "git-clone"
+              And user clicks on Add finally task again
+              And user clicks on Select task
+              And user selects "openshift-client"
+              And user clicks on Add finally task again
+              And user clicks on Select task
+              And user selects "kn"
+              And user fills task details
+              And user clicks on create
+             Then user sees finally tasks "git-clone", "openshift-client" and "kn" mentioned under "" section in the Pipeline details page
+
+
+        @regression, @manual
+        Scenario: When expression in the Pipeline Builder
+            Given user is at Pipeline Builder page
+              And user has chain of 4 tasks created in series
+             When user clicks on third task
+              And user navigates to When Expressions section
+              And user clicks on Add When Expressions
+             Then user sees a diamond shaped structure appear in front of third task
+              And user sees "Input", "Operator" and "Value" fields in When Expressions section
+              And user sees "Add Value", "Add When Expressions" and "Remove When Expressions" options
+
+
+        @regression
+        Scenario: Start pipeline with When expression in the Pipeline Builder
+            Given user is at Pipeline Builder page
+              And user has named pipeline as "pipeline-when-expression"
+              And user has tasks "git-cli", "git-clone" and "openshift-client" in series
+             When user clicks on third task
+              And user navigates to When Expressions section
+              And user clicks on Add When Expression
+              And user enters the input value as "$(tasks.git-clone.status)"
```

Comment:
You can't reference the status unless you're in a Finally Task.

From: https://tekton.dev/docs/pipelines/variables/

tasks.\<pipelineTaskName\>.status | The execution status of the specified pipelineTask, only available in finally tasks.
-- | --

Reply:
Does this apply to when expression section too?

Reply:
Yuppers. Please give this section a read-through to understand what is available for each property: https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#guard-task-execution-using-whenexpressions

They also have a list at the bottom of this section for use-cases they think WhenExpressions will be used for. Consider basing your test around one of those.

Reply:
I've updated the test scenario to include it in the finally task